benchmarks/test_nonlimber.py

```diff
@@ -0,0 +1,269 @@
+                                      nk_per_decade=200,
+                                      b1_pk_kind='linear',
+                                      bk2_pk_kind='linear')
+    ptc_lin.update_ingredients(cosmo)
+    ptt_g = [pt.PTNumberCountsTracer(b1=bias) for bias in tpar["b_g"]]
+    ptt_m = pt.PTMatterTracer()
+
+    tracers1 = {"gg": t_g, "gs": t_g, "ss": t_s}
+    tracers2 = {"gg": t_g, "gs": t_s, "ss": t_s}
+    truth = {"gg": tgg, "gs": tgs, "ss": tss}
+    errors = {"gg": err_gg, "gs": err_gs, "ss": err_ss}
+    indices = {"gg": indices_gg, "gs": indices_gs, "ss": indices_ss}
+    ptobj = {"ptc": ptc, "ptc_lin": ptc_lin, "ptt_g": ptt_g, "ptt_m": ptt_m}
+    return cosmo, ells, tracers1, tracers2, truth, errors, indices, ptobj
+
+
+@pytest.mark.parametrize("method", [None, "FKEM"])
+@pytest.mark.parametrize("cross_type", ["gg", "gs", "ss"])
+@pytest.mark.parametrize("pt_path", [False, True])
+def test_cells(set_up, method, cross_type, pt_path):
+    cosmo, ells, tracers1, tracers2, truth, errors, indices, ptobj = set_up
+    t0 = time.time()
+    chi2max = 0
+    biases = get_tracer_parameters()["b_g"]
+    for pair_index, (i1, i2) in enumerate(indices[cross_type]):
+        bias_fact = 1.0
+        p_of_k_a = p_of_k_a_lin = ccl.DEFAULT_POWER_SPECTRUM
+        if cross_type == "gg":
+            if pt_path:
+                p_of_k_a = ptobj['ptc'].get_biased_pk2d(
+                    ptobj['ptt_g'][i1],
+                    tracer2=ptobj['ptt_g'][i2])
+                p_of_k_a_lin = ptobj['ptc_lin'].get_biased_pk2d(
+                    ptobj['ptt_g'][i1],
+                    tracer2=ptobj['ptt_g'][i2])
+            else:
+                bias_fact = biases[i1] * biases[i2]
+        elif cross_type == "gs":
+            if pt_path:
+                p_of_k_a = ptobj['ptc'].get_biased_pk2d(
+                    ptobj['ptt_g'][i1],
+                    tracer2=ptobj['ptt_m'])
+                p_of_k_a_lin = ptobj['ptc_lin'].get_biased_pk2d(
+                    ptobj['ptt_g'][i1],
+                    tracer2=ptobj['ptt_m'])
+            else:
+                bias_fact = biases[i1]
+
+        if method is None:
+            cls = ccl.angular_cl(
+                cosmo,
+                tracers1[cross_type][i1],
+                tracers2[cross_type][i2],
+                ells,
+                l_limber=-1,
+                p_of_k_a=p_of_k_a,
+                p_of_k_a_lin=p_of_k_a_lin
+            )
+            l_limber = 0
+        else:
+            cls, meta = ccl.angular_cl(
+                cosmo,
+                tracers1[cross_type][i1],
+                tracers2[cross_type][i2],
+                ells,
+                p_of_k_a=p_of_k_a,
+                p_of_k_a_lin=p_of_k_a_lin,
+                l_limber='auto',
+                non_limber_integration_method=method,
+                return_meta=True
+            )
+            l_limber = meta['l_limber']
+        cls *= bias_fact
```

Comment:
Humm, why do you need this?

Reply:
the user may choose 'auto' for the non-limber code, which means it computes as limber after a convergence criteria is met. This allows the transitional ell mode to be returned to the user, which can be valuable for some science/systematics investigations.

Reply:
no, sorry, I was referring to the next line. Why do you need to multiply by the biases? Isn't that part of either the C_ell tracers or of the input Pks?

Reply:
This is addressed in lines 93-95 of the same file. It is there so that we can use the same approach for both Tracers and PTTracers in the same test function.